### PR TITLE
ROC-3629: rename isDefendantInCaseGuard -> onlyDefendantLinkedToClaimCanDoIt

### PR DESCRIPTION
**Pull request by yarpo**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-3629

### Change description
Refactoring

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/app/guards/onlyDefendantLinkedToClaimCanDoIt.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { User } from 'idam/user'
 import { Claim } from 'claims/models/claim'
 import { FeatureToggles } from 'utils/featureToggles'
 
-export class IsDefendantInCaseGuard {
```
Comment (geek-soft) [inline src/main/app/guards/onlyDefendantLinkedToClaimCanDoIt.ts]: Can we do the same to `IsClaimantInCaseGuard`?

Comment (yarpo) [reply src/main/app/guards/onlyDefendantLinkedToClaimCanDoIt.ts]: OK

**Comment by geek-soft** [comment]
@pragnesh-patel can you approve it. I think functional test pass should be enough to confirm it does not break anything.

**Comment by geek-soft** [comment]
E2E should be enough to test it.

**Pull request merged**